### PR TITLE
Fix null ref in persistence service

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs [hunk 1]
```diff
@@ -66,10 +66,13 @@ internal partial class AbstractSyntaxIndex<TIndex> : IObjectWritable
 
                 // attempt to load from persisted state
                 using var stream = await storage.ReadStreamAsync(documentKey, s_persistenceName, checksum, cancellationToken).ConfigureAwait(false);
-                using var gzipStream = new GZipStream(stream, CompressionMode.Decompress, leaveOpen: true);
-                using var reader = ObjectReader.TryGetReader(stream, cancellationToken: cancellationToken);
-                if (reader != null)
-                    return read(stringTable, reader, checksum);
+                if (stream != null)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs:69]: stream can be null when no data is stored.  didn't test the 'load first then save' scenario.

**Comment by CyrusNajmabadi** [comment]
@ryzngard @dibarbet ptal.

File: src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs [hunk 2]
```diff
@@ -189,6 +190,8 @@ private void Initialize(SqlConnection connection, CancellationToken cancellation
             EnsureTables(connection, Database.Main);
             EnsureTables(connection, Database.WriteCache);
 
+            LoadExistingStringIds(connection);
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs:193]: What's this change for?

**Comment by dibarbet** [comment]
@CyrusNajmabadi what was the original null ref?  the stream being null?

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi what was the original null ref? the stream being null?

Yes.

File: src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs:193]: Will doc.  But something i noticed while testing.  we were hitting the DB heavily on the initial loads reading in all the string->id pairs, one at a time as different reads hit different strings.

This is much more efficient as we can just stream them all in in one swell foop at initialization time.

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs:193]: but i can leave this out of this PR :)

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs]: removed.  will add in followup.

Comment (ryzngard) [reply src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs]: Please do a separate pr 🙏 will make it easier to diagnose any perf issues

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs]: yup yup.  it's removed! :)

File: src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs [hunk 3]
```diff
@@ -45,6 +45,7 @@ internal sealed partial class SQLitePersistentStorage : AbstractPersistentStorag
 
         private readonly string _insert_into_string_table_values_0 = $"insert into {StringInfoTableName}({DataColumnName}) values (?)";
         private readonly string _select_star_from_string_table_where_0_limit_one = $"select * from {StringInfoTableName} where ({DataColumnName} = ?) limit 1";
+        private readonly string _select_star_from_string_table = $"select * from {StringInfoTableName}";
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs:48]: Not, remove as well

**Review by ryzngard — APPROVED** [review]

**Review by dibarbet — APPROVED** [review]

File: src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs:48]: will do!

**Comment by dibarbet** [comment]
> > @CyrusNajmabadi what was the original null ref? the stream being null?
> 
> Yes.

hmm for some reason your original comment on it didn't show up when I first loaded the PR.  shows up now though...

**Pull request merged**